source/entities/GameEntity.cpp

```diff
@@ -616,4 +615,20 @@ void GameEntity::notifyFightPlayer(Tile* tile)
     getGameMap()->playerIsFighting(getSeat()->getPlayer(), tile);
 }
 
+void GameEntity::setParentNodeDetach(uint32_t mask, bool value)
+{
+    // We save the current attach  state
+    bool oldState = (mEntityParentNodeAttach == EntityParentNodeAttach::ATTACHED);
+    // We compute the new attach state
+    mEntityParentNodeAttach = (value ? mEntityParentNodeAttach | mask : mEntityParentNodeAttach & ~mask);
+
+    // If the attach state changed, we do what is needed
+    bool newState = (mEntityParentNodeAttach == EntityParentNodeAttach::ATTACHED);
+    if(oldState == newState)
+        return;
```

Comment:
It does it means : to get here I must first meet condition oldState!=newState which must be deduced in your case. So if I am in RenderManager::getSingleton().rrAttachEntity(this); I explicitly know that newState is true and OldState is false.